src/app/core/session/session.service.provider.ts

```diff
@@ -35,54 +38,48 @@ import { LocalSession } from "./session-service/local-session";
 export function sessionServiceFactory(
   alertService: AlertService,
   loggingService: LoggingService,
-  entitySchemaService: EntitySchemaService,
-  httpClient: HttpClient
+  httpClient: HttpClient,
+  demoDataService: DemoDataService,
+  database: Database,
+  analyticsService: AnalyticsService
 ): SessionService {
+  const pouchDatabase = database as PouchDatabase;
   let sessionService: SessionService;
-  switch (AppConfig.settings.session_type) {
-    case SessionType.local:
-      sessionService = new LocalSession(
-        PouchDatabase.createWithIndexedDB(
-          AppConfig.settings.database.name,
-          loggingService
-        )
-      );
-      break;
-    case SessionType.synced:
-      sessionService = new SyncedSessionService(
-        alertService,
-        loggingService,
-        entitySchemaService,
-        httpClient
-      );
-      break;
-    default:
-      sessionService = new LocalSession(
-        PouchDatabase.createWithInMemoryDB(
-          AppConfig.settings.database.name,
-          loggingService
-        )
-      );
-      break;
+  if (AppConfig.settings.demo_mode === true) {
```

Comment:
Do we have to tightly couple demo data and sessions? Separation of concerns keeps things easier to understand and change. And it has proven useful to generate demo data into a local session as well.